chain/store_test.go

```diff
@@ -406,94 +182,90 @@ func TestGetTipSetState(t *testing.T) {
 // Tipsets can be retrieved by parent key (all block cids of parents).
 func TestGetByParent(t *testing.T) {
 	tf.UnitTest(t)
-	dstP := initDSTParams()
 
 	ctx := context.Background()
-	initStoreTest(ctx, t, dstP)
-	chain := newChainStore(dstP)
-
-	requirePutTestChain(t, chain, dstP)
-	pkg := types.TipSetKey{} // empty cid set is dstP.genesis pIDs
-	pk1 := dstP.genTS.Key()
-	pk2 := dstP.link1.Key()
-	pk3 := dstP.link2.Key()
-	pk4 := dstP.link3.Key()
-
-	gotG := requireGetTsasByParentAndHeight(t, chain, pkg, uint64(0))
-	got1 := requireGetTsasByParentAndHeight(t, chain, pk1, uint64(1))
-	got2 := requireGetTsasByParentAndHeight(t, chain, pk2, uint64(2))
-	got3 := requireGetTsasByParentAndHeight(t, chain, pk3, uint64(3))
-	got4 := requireGetTsasByParentAndHeight(t, chain, pk4, uint64(6)) // two null blocks in between 3 and 4!
-
-	assert.Equal(t, dstP.genTS, gotG[0].TipSet)
-	assert.Equal(t, dstP.link1, got1[0].TipSet)
-	assert.Equal(t, dstP.link2, got2[0].TipSet)
-	assert.Equal(t, dstP.link3, got3[0].TipSet)
-	assert.Equal(t, dstP.link4, got4[0].TipSet)
-
-	assert.Equal(t, dstP.genStateRoot, gotG[0].TipSetStateRoot)
-	assert.Equal(t, dstP.link1State, got1[0].TipSetStateRoot)
-	assert.Equal(t, dstP.link2State, got2[0].TipSetStateRoot)
-	assert.Equal(t, dstP.link3State, got3[0].TipSetStateRoot)
-	assert.Equal(t, dstP.link4State, got4[0].TipSetStateRoot)
+	builder := chain.NewBuilder(t, address.Undef)
+	genTS := builder.NewGenesis()
+	r := repo.NewInMemoryRepo()
+	cs := newChainStore(r, genTS.At(0).Cid())
+
+	// Construct test chain data
+	link1 := builder.AppendOn(genTS, 2)
+	link2 := builder.AppendOn(link1, 3)
+	link3 := builder.AppendOn(link2, 1)
+	link4 := builder.BuildOn(link3, 2, func(bb *chain.BlockBuilder, i int) { bb.IncHeight(2) })
```

Comment:
I could abstract out the frequently repeated lines from 187 to 196 if reviewers prefer.

Reply:
No need IMO, but you could pull a lot of these simpler tests into `t.Run()` sharing the same builder/links and chain.